packages/circuits/src/defaultProver.ts

```diff
@@ -0,0 +1,40 @@
+import path from 'path'
+import { Circuit } from '.'
+import * as snarkjs from 'snarkjs'
+import { SnarkProof, SnarkPublicSignals } from '@unirep/crypto'
+
+const buildPath = '../zksnarkBuild'
+
+export const defaultProver = {
```

Comment:
I was trying to avoid a circular import there. I really intended the `Prover` interface to be simple enough that it doesn't need a full class. Any object with the properties `verifyProof` and `genProofAndPublicSignals` is a valid Prover.

I was going to add `export const defaultProver = {...} as Prover` but it's redundant. The object already has the needed properties so it can be passed as a `Prover`.

I think this is a matter of preference - either the class or the object syntax is valid. I didn't use the class syntax because it would never be instantiated, each function would need to be static. It's an alright approach I just didn't do it here.

Reply:
Thanks for your explain!